pictorus-internal/src/loggers/csv_logger.rs

```diff
@@ -68,77 +55,76 @@ impl Logger for CsvLogger {
-            let mut field = fields.next();
-            while let Some(ref field_inner) = field {
-                match field_inner.1.begin() {
-                    Fragment::Null => {} // This indicates a None value, we just have it be empty
-                    Fragment::Seq(_) => {
-                        // This indicates an array, need to add `"` to avoid parsers splitting the array up incorrectly
-                        sample.push('"');
-                        sample.push_str(&miniserde::json::to_string(field_inner.1));
-                        sample.push('"');
-                    }
-                    Fragment::Map(_) => {
-                        // We don't support this or expect to see it
-                        panic!("Unsupported data format for CSV samples");
-                    }
-                    Fragment::Bool(_)
-                    | Fragment::F64(_)
-                    | Fragment::I64(_)
-                    | Fragment::U64(_)
-                    | Fragment::Str(_) => {
-                        // Simple scalar values don't need any special formatting
-                        sample.push_str(&miniserde::json::to_string(field_inner.1));
-                    }
+    let json = serde_json::to_value(data).unwrap();
+    if let Some(json_map) = json.as_object() {
+        for (_, value) in json_map {
+            match value {
+                serde_json::Value::Null => {}
+                serde_json::Value::Bool(_) => {
+                    // This indicates a boolean value, we just have it be empty
+                    sample.push_str(&serde_json::to_string(value).unwrap());
                 }
-                field = fields.next();
-                if field.is_some() {
-                    // Add a comma between fields
-                    sample.push(',');
+                serde_json::Value::Number(number) => {
+                    // This indicates a number value, we just have it be empty
+                    sample.push_str(&number.to_string());
+                }
+                serde_json::Value::String(_) => {
+                    // This indicates a string value, we just have it be empty
+                    sample.push_str(&serde_json::to_string(value).unwrap());
+                }
+                serde_json::Value::Array(values) => {
+                    sample.push('"');
+                    sample.push_str(&serde_json::to_string(values).unwrap());
+                    sample.push('"');
+                }
+                serde_json::Value::Object(_map) => {
+                    // We don't support this or expect to see it
+                    panic!("Unsupported data format for CSV samples");
                 }
             }
-        }
-        _ => {
-            panic!("Unsupported data format for CSV samples");
+            sample.push(',');
         }
     }
+
+    if sample.ends_with(',') {
+        sample.pop(); // Remove the trailing comma
+    }
```

Comment:
same super nit on this being a bit of a code smell to me (although not messing with manually calling `next()` on an iterator is definitely an improvement